domain/src/test/java/org/oppia/domain/oppialogger/exceptions/ExceptionsControllerTest.kt

```diff
@@ -0,0 +1,350 @@
+    assertThat(exception.cause?.message).isEqualTo(exceptionThrown.cause?.message)
+    assertThat(exception.cause?.stackTrace).isEqualTo(exceptionThrown.cause?.stackTrace)
+    assertThat(exceptionLog.exceptionType).isEqualTo(ExceptionType.NON_FATAL)
+  }
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logException_Fatal_withNoNetwork_logsToCacheStore() {
+    networkConnectionUtil.setCurrentConnectionStatus(NetworkConnectionUtil.ConnectionStatus.NONE)
+    val exceptionThrown = Exception("TEST MESSAGE", Throwable("TEST"))
+    exceptionsController.logFatalException(exceptionThrown, TEST_TIMESTAMP)
+
+    val cachedExceptions = exceptionsController.getExceptionLogs()
+    cachedExceptions.observeForever(mockOppiaExceptionLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    verify(mockOppiaExceptionLogsObserver, atLeastOnce())
+      .onChanged(oppiaExceptionLogsResultCaptor.capture())
+
+    val exceptionLog = oppiaExceptionLogsResultCaptor.value.getOrThrow().getExceptionLog(0)
+    val exception = exceptionLogToException(exceptionLog)
+    assertThat(exception.message).isEqualTo(exceptionThrown.message)
+    assertThat(exception.stackTrace).isEqualTo(exceptionThrown.stackTrace)
+    assertThat(exception.cause?.message).isEqualTo(exceptionThrown.cause?.message)
+    assertThat(exception.cause?.stackTrace).isEqualTo(exceptionThrown.cause?.stackTrace)
+    assertThat(exceptionLog.exceptionType).isEqualTo(ExceptionType.FATAL)
+  }
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  @Test
+  fun testController_logExceptions_exceedLimit_checkCorrectEviction() {
+    networkConnectionUtil.setCurrentConnectionStatus(NetworkConnectionUtil.ConnectionStatus.NONE)
+
+    exceptionsController.logFatalException(Exception("TEST1", Throwable("ONE")), 1556094120000)
+    exceptionsController.logNonFatalException(Exception("TEST2", Throwable("TWO")), 1556094110000)
+    exceptionsController.logFatalException(Exception("TEST3", Throwable("THREE")), 1556094100000)
+    exceptionsController.logFatalException(Exception("TEST4", Throwable("FOUR")), 1556094000000)
+
+    val cachedExceptions = exceptionsController.getExceptionLogs()
+    cachedExceptions.observeForever(mockOppiaExceptionLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    verify(mockOppiaExceptionLogsObserver, atLeastOnce())
+      .onChanged(oppiaExceptionLogsResultCaptor.capture())
+
+    val exceptionOne = oppiaExceptionLogsResultCaptor.value.getOrThrow().getExceptionLog(0)
+    val exceptionTwo = oppiaExceptionLogsResultCaptor.value.getOrThrow().getExceptionLog(1)
+    val cacheStoreSize = oppiaExceptionLogsResultCaptor.value.getOrThrow().exceptionLogList.size
+
+    assertThat(cacheStoreSize).isEqualTo(2)
+    // In this case, 3 FATAL and 1 NON-FATAL exception was logged. So while pruning, none of the retained logs should have NON-FATAL exception type.
```

Comment:
Here & everywhere else in this PR: please refer to 'fatal' and 'non-fatal' as standard words rather than specifically capitalizing them. We don't do this for any other terms in the codebase, and it's throwing me a bit off.

Reply:
Done.